src/content/api.js

```diff
@@ -86,9 +88,11 @@ const getEntitiesFromText = textData => new Promise((resolve, reject) => {
     Http.send();
 });
 
-export const getUniqueLocationsFromText = textData => new Promise(resolve => getEntitiesFromText(textData)
+export const getUniqueLocationsFromText = textData => getEntitiesFromText(textData)
```

Comment:
yes. the .then() only works because we are returning a promise. Sometimes the thing we are returning is already returning a promise itself, but that is not the case with this primitive Http api. I'd like to move to request or axios library soon to remove this need but for now I do believe it is needed

Reply:
So if filterDuplicates does not return a Promise, won't we need to return a promise for getUniqueLocationsFromText? Or will the Promise from getEntitiesFromText carry over?

Reply:
yeah the promise from getEntitiesFromText carries over